message/validation/consensus_state.go

```diff
@@ -14,26 +17,22 @@ type consensusID struct {
 
 // consensusState keeps track of the signers for a given public key and role.
 type consensusState struct {
-	signers map[spectypes.OperatorID]*SignerState
-	mu      sync.Mutex
+	state    map[spectypes.OperatorID]*treemap.Map // TODO: use *treemap.Map[phase0.Slot, *SignerState] after updating to Go 1.21
+	maxSlots phase0.Slot
+	mu       sync.Mutex
 }
 
-// GetSignerState retrieves the state for the given signer.
-// Returns nil if the signer is not found.
-func (cs *consensusState) GetSignerState(signer spectypes.OperatorID) *SignerState {
+func (cs *consensusState) Get(signer spectypes.OperatorID) *treemap.Map {
```

Comment:
function name should represent that it creates the item if it doesn't exist, in the name or at least comment.

Reply:
I agree, thanks